integration-tests/index.test.js

```diff
@@ -660,3 +612,241 @@ query ntee {
+
+    test('add pdf for active user', async () => {
+      const { uri, instance, db } = await createServerInstance();
+
+      const client = new GraphQLClient(uri, {
+        headers: {
+          'X-Auth-Token': 'good user',
+        },
+      });
+
+      const {
+        organization: { uuid: orgUuid },
+      } = await client.request(
+        `
+query org {
+  organization(id: 93) {
+    uuid
+  }
+}`
+      );
+
+      const res = await client.request(
+        `
+mutation pdf {
+  addPdf(input: {
+    organization: "${orgUuid}"
+    url: "bar"
+    done: true
+    year: 12
+    user: "${user.uuid}"
+  }) {
+    url
+    year
+    done
+    uuid
+    organization {uuid}
+  }
+}
+`
+      );
+
+      expect(res.addPdf).toEqual({
+        url: 'bar',
+        year: 12,
+        done: true,
+        uuid: res.addPdf.uuid,
+        organization: { uuid: orgUuid },
+      });
+
+      thisUserPdf = res.addPdf;
+
+      instance.close();
+    });
+
+    test('add pdf for another user', async () => {
+      const { uri, instance, db } = await createServerInstance();
+
+      const client = new GraphQLClient(uri, {
+        headers: {
+          'X-Auth-Token': 'good user',
+        },
+      });
+
+      const {
+        organization: { uuid: orgUuid },
+      } = await client.request(
+        `
+query org {
+  organization(id: 93) {
+    uuid
+  }
+}`
+      );
+
+      const res = await client.request(
+        `
+mutation pdf {
+  addPdf(input: {
+    organization: "${orgUuid}"
+    url: "baz"
+    done: false
+    year: 13
+    user: "${user2.uuid}"
+  }) {
+    url
+    year
+    done
+    uuid
+    organization {uuid}
+  }
+}
+`
+      );
+
+      expect(res.addPdf).toEqual({
+        url: 'baz',
+        year: 13,
+        done: false,
+        uuid: res.addPdf.uuid,
+        organization: { uuid: orgUuid },
+      });
+
+      anotherUserPdf = res.addPdf;
+
+      instance.close();
+    });
+
+    test('list active user pdfs', async () => {
+      const { uri, instance, db } = await createServerInstance();
+
+      const client = new GraphQLClient(uri, {
+        headers: {
+          'X-Auth-Token': 'good user',
+        },
+      });
+
+      const res = await client.request(
+        `
+query pdfs {
+  pdfs(limitToCurrentUser: true) {
+    uuid
+  }
+}
+`
+      );
+
+      expect(res.pdfs.length).toBe(1);
+      expect(res.pdfs[0].uuid).toEqual(thisUserPdf.uuid);
+
+      instance.close();
+    });
+
+    test('list all pdfs', async () => {
+      const { uri, instance, db } = await createServerInstance();
+
+      const client = new GraphQLClient(uri, {
+        headers: {
+          'X-Auth-Token': 'good user',
+        },
+      });
+
+      const res = await client.request(
+        `
+query pdfs {
+  pdfs(limitToCurrentUser: false) {
+    uuid
+  }
+}
+`
+      );
+
+      expect(res.pdfs.length).toBe(2);
+      expect(res.pdfs[0].uuid).toEqual(thisUserPdf.uuid);
+      expect(res.pdfs[1].uuid).toEqual(anotherUserPdf.uuid);
```

Comment:
Wait, does it not randomize the test order? 
In general I would expect each test to be self-contained in terms of setup and teardown.

Reply:
jest executes the tests within a suite sequentially - https://github.com/facebook/jest/issues/4386 - the files themselves will be run in a random order though

Reply:
Ah I'm so used to rails tests, which randomize order to enforce test isolation and catch random order-of-operations bugs.